### PR TITLE
Add 2023 Vermont Tax Rules

### LINKED ISSUE #4200: Add 2023 Vermont Tax Rules
2023 Income Tax Form Instructions: https://tax.vermont.gov/sites/tax/files/documents/Income-Booklet-2023.pdf

### PR DESCRIPTION
**Pull request by Kiki-67**
Fixes #4200

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: changelog

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml [hunk 2]
```diff
@@ -10,22 +10,29 @@ metadata:
       href: https://tax.vermont.gov/sites/tax/files/documents/Income-Booklet-2021.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:10]: add to the metadata:
`uprating: gov.irs.uprating`

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: could you explain the changes in the issue, what changed and to what amount

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/head_of_household.yaml [hunk 3]
```diff
@@ -12,6 +12,8 @@ metadata:
       href: https://tax.vermont.gov/sites/tax/files/documents/RateSched-2021.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/head_of_household.yaml:12]: add to the metadata:
`uprating: gov.irs.uprating`

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml [hunk 4]
```diff
@@ -12,6 +12,8 @@ metadata:
       href: https://tax.vermont.gov/sites/tax/files/documents/RateSched-2021.pdf#page=1
     - title: Vermont 2022 Income Tax Return Booklet Forms and Instructions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml:13]: add to the metadata:
uprating: gov.irs.uprating

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/separate.yaml [hunk 5]
```diff
@@ -12,6 +12,8 @@ metadata:
       href: https://tax.vermont.gov/sites/tax/files/documents/RateSched-2021.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/separate.yaml:12]: add to the metadata:
uprating: gov.irs.uprating

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/single.yaml [hunk 6]
```diff
@@ -12,6 +12,8 @@ metadata:
       href: https://tax.vermont.gov/sites/tax/files/documents/RateSched-2021.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/single.yaml:12]: add to the metadata:
uprating: gov.irs.uprating

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/widow.yaml [hunk 7]
```diff
@@ -12,6 +12,8 @@ metadata:
       href: https://tax.vermont.gov/sites/tax/files/documents/RateSched-2021.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/widow.yaml:12]: add to the metadata:
uprating: gov.irs.uprating

File: policyengine_us/parameters/gov/states/vt/tax/income/exemption/personal.yaml [hunk 8]
```diff
@@ -14,3 +15,5 @@ metadata:
       href: https://tax.vermont.gov/sites/tax/files/documents/IN-111-2022.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/exemption/personal.yaml:15]: add to the metadata:
uprating: gov.irs.uprating

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/additional.yaml [hunk 9]
```diff
@@ -11,6 +11,8 @@ metadata:
      # Legal code specifies $1,000 with inflation adjustments.
     - title: Legal Code Titl. 32 V.S.A. § 5811(21)(C)(iii)
       href: https://legislature.vermont.gov/statutes/section/32/151/05811
-
+    - title: 2023 VERMONT Income Tax Return Form Instructions-Taxable Income Line 4
+      href: https://tax.vermont.gov/sites/tax/files/documents/Income%20Booklet-2022.pdf#page=10 
 values:
   2021-01-01: 1_050
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/additional.yaml:17]: could you check 2022 value if it changed?

If so please add:
add to the metadata:
uprating: gov.irs.uprating

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml [hunk 10]
```diff
@@ -4,6 +4,7 @@ metadata:
   threshold_unit: currency-USD
   rate_unit: /1
   label: Vermont State income tax rates for joint filers
+  uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml:7]: we have to add the uprating here to each threshold individually (we dont want the rate to be updated each year)
- see the CA PR files:
https://github.com/PolicyEngine/policyengine-us/pull/3770/files#diff-ab9344c8a2727114a1c0a5e36747a11cf0ad22885047cc17c26aba08d552645a

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml:7]: same for all marginal rate or single amount parameters

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml [hunk 11]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add 2023 Vermont Tax Rules.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - 2023 Vermont Tax Rules.
```

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml (hunk 10 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml:7]: Could you check whether the values are rounded or not?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml [hunk 12]
```diff
@@ -3,29 +3,37 @@ metadata:
   label: Vermont standard deduction
   period: year
   unit: currency-USD
+  uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:6]: is this rounded?
If so we need to add some extra uprating metadata, see DC:
https://github.com/PolicyEngine/policyengine-us/pull/4306

Need to double check in legal code

File: policyengine_us/parameters/gov/states/vt/tax/income/exemption/personal.yaml [hunk 13]
```diff
@@ -2,10 +2,12 @@ description: Vermont provides the following personal exemption amount.
 values:
   2021-01-01: 4_350
   2022-01-01: 4_500
+  2023-01-01: 4_850
 metadata:
   unit: currency-USD
   period: year
   label: Vermont personal exemption amount
+  uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/exemption/personal.yaml:10]: rounding? same as above

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/head_of_household.yaml [hunk 14]
```diff
@@ -12,23 +12,37 @@ metadata:
...
       2021-01-01: 0.0335
   - threshold:
-      2021-01-01: 54_850
-      2022-01-01: 56_500
+      values:
+        2021-01-01: 54_850
+        2022-01-01: 56_500
+        2023-01-01: 60_850
+      metadata:
+        uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/head_of_household.yaml:28]: rounding? same as above

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/separate.yaml [hunk 15]
```diff
@@ -12,23 +12,37 @@ metadata:
...
       2021-01-01: 0.0335
   - threshold:
-      2021-01-01: 34_200
-      2022-01-01: 35_225
+      values:
+        2021-01-01: 34_200
+        2022-01-01: 35_225
+        2023-01-01: 37_925
+      metadata:
+        uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/separate.yaml:28]: rounding? same as above

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml (hunk 12 above)
Comment (Kiki-67) [reply policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:6]: didn't find documents saying the number is rounded.

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:6]: I dont think there is rounding based on:
image

File: policyengine_us/tests/policy/baseline/gov/states/vt/tax/income/vt_normal_income_tax.yaml [hunk 16]
```diff
@@ -129,3 +129,13 @@
   output:
     # .0335*35255 + .066*(85150-35255) + .076*(129750-85150) + .0875*(150000-129750) = 9636.5625
     vt_normal_income_tax: 9_636.5625
+
+- name: 14 Joint filer, taxable income is $75.25K in 2023.
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/vt/tax/income/vt_normal_income_tax.yaml:133]: tests fail now

File: policyengine_us/tests/policy/baseline/gov/states/vt/tax/income/vt_normal_income_tax.yaml [hunk 17]
```diff
@@ -129,3 +129,13 @@
...
+  period: 2023
+  input:
+    vt_taxable_income: 75_250
+    filing_status: JOINT
+    state_code: VT
+  output:
+    # .0335*75000 + .0335*(75250-75000)
+    vt_normal_income_tax: 2520.875
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/vt/tax/income/vt_normal_income_tax.yaml:141]:
```suggestion
    vt_normal_income_tax: 2_520.875
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml (hunk 12 above)
Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:6]: please file an issue to check on this. the numbers are rounded so we may want to check with the revenue agency to see what other part of the law defines it

**Comment by PavelMakarchuk** [comment]
@Kiki-67 I believe this is caused because we are uprating the 2022 parameters if though the uprating should effectively start in 2023 
For this please take a look at [this](https://github.com/PolicyEngine/policyengine-core/pull/173) core PR and implement the new metadata for the uprating 
There is an example in the PR

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml [hunk 18]
```diff
@@ -12,23 +12,47 @@ metadata:
...
+      href: https://tax.vermont.gov/sites/tax/files/documents/RateSched-2023.pdf#page=1
 brackets:
   - threshold:
       2021-01-01: 0
     rate:
       2021-01-01: 0.0335
   - threshold:
-      2021-01-01: 68_400
-      2022-01-01: 70_450
+      values:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml:23]: consolidate brackets with the same rates

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/widow.yaml [hunk 19]
```diff
@@ -4,6 +4,7 @@ metadata:
   threshold_unit: currency-USD
   rate_unit: /1
   label: Vermont income tax rates widow(er) filers
+  uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/widow.yaml:7]:
```suggestion
```

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/single.yaml [hunk 20]
```diff
@@ -4,6 +4,7 @@ metadata:
   threshold_unit: currency-USD
   rate_unit: /1
   label: Vermont income tax rates single filers
+  uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/single.yaml:7]:
```suggestion
```

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/single.yaml [hunk 21]
```diff
@@ -20,15 +23,24 @@ brackets:
   - threshold:
       2021-01-01: 40_950
       2022-01-01: 42_150
+      2023-01-01: 45_400
+    # metadata:
+    #   uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/single.yaml:28]: uncomment

File: policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml [hunk 22]
```diff
@@ -20,15 +22,24 @@ brackets:
   - threshold:
       2021-01-01: 68_400
       2022-01-01: 70_450
+      2023-01-01: 75_850
+    metadata:
+      uprating: gov.irs.uprating
     rate:
       2022-01-01: 0.066
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/rates/joint.yaml:29]:
```suggestion
      2021-01-01: 0.066
```

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml (hunk 12 above)
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:6]:
```suggestion
```

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml [hunk 23]
```diff
@@ -3,29 +3,37 @@ metadata:
...
+    - title: 2023 VERMONT Income Tax Return Form Instructions-Taxable Income Line 4
+      href: https://tax.vermont.gov/sites/tax/files/documents/Income-Booklet-2021.pdf#page=10
       # Legal code specifies amount with inflation adjustments..
     - title: Legal Code Titl. 32 V.S.A. § 5811(21)(C)(ii) 
       href: https://legislature.vermont.gov/statutes/section/32/151/05811 
 
 JOINT:
   2021-01-01: 12_700
   2022-01-01: 13_050
+  2023-01-01: 14_050
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:23]:
```suggestion
  2023-01-01: 14_050
  metadata:
      uprating: gov.irs.uprating
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/base.yaml:23]: for each filing status

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/additional.yaml [hunk 24]
```diff
@@ -11,6 +12,8 @@ metadata:
      # Legal code specifies $1,000 with inflation adjustments.
     - title: Legal Code Titl. 32 V.S.A. § 5811(21)(C)(iii)
       href: https://legislature.vermont.gov/statutes/section/32/151/05811
-
+    - title: 2023 VERMONT Income Tax Return Form Instructions-Taxable Income Line 4
+      href: https://tax.vermont.gov/sites/tax/files/documents/Income%20Booklet-2022.pdf#page=10
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/vt/tax/income/deductions/standard/additional.yaml:16]: this is the 2022 tax form reference, please double check all references

**Review by PavelMakarchuk — APPROVED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**